dev/bench/htmloutput.ml

```diff
@@ -30,21 +30,44 @@ let htmlescape =
 let percentage ~max:m v =
   Q.to_float Q.(v * of_int 100 / m)
 
+let pp_words ~need_comma which w =
+  if w = "0 w" then need_comma, ""
+  else
+    true, (if need_comma then ", " else "")^(String.sub w 0 (String.length w - 1))^which^" w"
+
+let pp_collect ~need_comma which c =
+  if c = 0 then need_comma, ""
+  else
+    true, Printf.sprintf "%s%d %s %s"
+      (if need_comma then ", " else "") c which
+      (if c = 1 then "collection" else "collections")
+
+let pp_memory ch = function
+  | None -> ()
+  | Some {major_words; minor_words; major_collect; minor_collect} ->
+    (* need_comma <-> prefix is nontrivial *)
+    let need_comma, major_words = pp_words ~need_comma:false "major" major_words in
+    let need_comma, minor_words = pp_words ~need_comma "minor" minor_words in
+    let need_comma, major_collect = pp_collect ~need_comma "major" major_collect in
+    let need_comma, minor_collect = pp_collect ~need_comma "minor" minor_collect in
+    if need_comma then
+      Printf.fprintf ch " (%s%s%s%s)" major_words minor_words major_collect minor_collect
```

Comment:
swapping major and minor would produce nicer alignment
typically you would get
~~~
Time1: Xs (X minor w, X major w, X minor coll, X major coll)
Time2: Xs (X minor w, X major w, X minor coll)
~~~
if only 1 side has a major collection (swapping minor/major words may have similar benefits if rarer and also keeps it consistent with the order for collection printing)